go/vt/servenv/grpc_static_auth_plugin.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servenv
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	log "github.com/golang/glog"
+)
+
+var (
+	credsFile = flag.String("grpc_static_auth_plugin_file", "", "JSON File to read the users/passwords from.")
+	// StaticAuthClientCreds implements client interface to be able to WithPerRPCCredentials
+	_ credentials.PerRPCCredentials = (*StaticAuthClientCreds)(nil)
+)
+
+// StaticAuthClientCreds holder for client credentials
+type StaticAuthClientCreds struct {
+	Username string
+	Password string
+}
+
+// StaticAuthConfigEntry holder for server side credentials. Current implementation matches the
+// the one from the client but this will change in the future as we hooked this pluging into ACL
+// features.
+type StaticAuthConfigEntry struct {
+	Username string
+	Password string
+	// TODO (@rafael) Add authorization parameters
+}
+
+func (c *StaticAuthClientCreds) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{
+		"username": c.Username,
+		"password": c.Password,
+	}, nil
+}
+
+func (c *StaticAuthClientCreds) RequireTransportSecurity() bool {
```

Comment:
Can you add a comment as to why you're setting this to false.

Also FWIW once we try to move upstream _all_ these public functions will need comments as per golint.